application/lib/editor.dart

```diff
@@ -338,41 +338,40 @@ Widget _pad(Widget child) => Padding(
   child: child,
 );
 
-Widget _makeAtomSlot(Set<String> classes, Atom value, Atom parent, ValueSetter<Atom> onChanged, {
-  @required bool needsTree,
-  @required bool needsDifferent,
+Widget _makeAtomSlot(Set<String> classes, Atom? value, Atom parent, ValueSetter<Atom?>? onChanged, {
+  required bool needsTree,
+  required bool needsDifferent,
 }) {
-  assert(parent != null);
-  bool _ok(Atom atom) => (!needsTree || parent.canAddToTree(atom))
+  bool _ok(Atom? atom) => (!needsTree || parent.canAddToTree(atom!))
                       && (!needsDifferent || parent != atom);
   return DragTarget<Atom>(
-    onWillAccept: (Atom atom) => classes.contains(atom.className) && _ok(atom),
-    onAccept: (Atom atom) {
-      if (_ok(atom))
-        onChanged(atom);
+    onWillAcceptWithDetails: (DragTargetDetails<Atom>? atom) => classes.contains(atom!.data.className) && _ok(atom.data),
```

Comment:
can we call this argument `details` rather than `atom`?

Reply:
fixed